tests/ludwig/encoders/test_image_encoders.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+import torch
+
+from ludwig.encoders.image_encoders import Stacked2DCNN, ResNetEncoder,\
+    MLPMixerEncoder
+
+
+@pytest.mark.parametrize(
+    'img_height,img_width,num_conv_layers,first_in_channels', [(224, 224, 5, 3)]
+)
+def test_stacked2d_cnn(
+        img_height: int,
+        img_width: int,
+        num_conv_layers: int,
+        first_in_channels: int
+):
+    
+    stacked_2d_cnn = Stacked2DCNN(
+        img_height, img_width, num_conv_layers=num_conv_layers, 
+        first_in_channels=first_in_channels)
+    inputs = torch.rand(2, first_in_channels, img_height, img_width)
+    outputs = stacked_2d_cnn(inputs)
+    assert outputs['encoder_output'].shape[1:] == stacked_2d_cnn.output_shape
+
+
+@pytest.mark.parametrize('img_height,img_width', [(224, 224)])
+def test_resnet_encoder(img_height: int, img_width: int):
+    resnet = ResNetEncoder(img_height, img_width)
+    inputs = torch.rand(2, 3, img_height, img_width)
+    outputs = resnet(inputs)
+    assert outputs['encoder_output'].shape[1:] == resnet.output_shape
```

Comment:
Just an observation...`reset.output_shape` is set by the fully connected layer `output_shape`.  The `output_shape` for fully connected is calculated by using the default `LudwigModule.output_shape` property, which sends a "fake" tensor through the `forward()` method.  

It seems that the `assert` test is comparing the outputs that are derived from the same mechanism.  So if an error occurred in constructing the output tensor from the fully connected layer, then the assertion would not catch the discrepancy.   If this was a math proof, isn't this like assuming the consequence. :-)

From what I can tell, this comment applies to the other tests for image encoders as well.

OTOH,  in the assert test for shapes in the 2D convolutional tests one side is derived directly from the parameters of the constructors, which is independent of the processing in the `forward()` methods.

Reply:
@jimthompson5802 good catch, this set of tests won't catch the errors from the FC forward! IMO, the encoder output tests the the full forward pass of the encoder, while the resnet output property is the independently computed output shape of the module. For the FC layer, the output shape is computed by calling forward only on the FC layer, so errors that originate in piping the Resnet module to the FC layer will be picked up. However, I'm open to suggestions on how to catch errors in FC output tensors.

Reply:
That's reasonable, i.e., if FC has the independent verification, then anything using it can assume proper operation and make use of its results

For the sequence encoders, the approach I took was to use the relevant values of the encoder parameters for the output shape assertion test.  [Here is an example.](https://github.com/jimthompson5802/ludwig/blob/f09358e4530cdd3871932c97db0e1c1b922c159c/tests/integration_tests/test_sequence_encoders.py#L90-L126)

I think either approach should meet the requirement.

Reply:
FYI I'm mergin the PR, if we want to do modifications tothe tests we could do them in a follow up PR for both seq and image tests